front/src/common/Map/Layers/InfraObjectLayers/OperationalPoints.tsx

```diff
@@ -14,14 +19,44 @@ type OperationalPointsProps = {
   infraID: number | undefined;
   operationnalPointId?: string;
   highlightedArea?: Geometry;
+  highlightedOperationalPoints?: number[];
 };
 
+function getColorByHighlighted(data: {
+  highlightedArea?: Geometry;
+  highlightedOperationalPoints?: number[];
+  inColor: string;
+  outColor: string;
+}): DataDrivenPropertyValueSpecification<ColorSpecification> {
```

Comment:
nit: we could avoid using data (same for the other function)

```suggestion
function getColorByHighlighted({
  highlightedArea?: Geometry;
  highlightedOperationalPoints?: number[];
  inColor: string;
  outColor: string;
}): DataDrivenPropertyValueSpecification<ColorSpecification> {
```

Reply:
It's not possible to define at the same time a type and the variable names.

What I can do is to remove the object inside the function parameters, but then I need to know the order of the function's parameters, specially with the two optional ones

Reply:
Yeah we could probably do 
```
getColorByHighlighted({
  highlightedArea,
  highlightedOperationalPoints,
  inColor,
  outColor
}: {
  highlightedArea?: Geometry;
  highlightedOperationalPoints?: number[];
  inColor: string;
  outColor: string;
})
```
But it's a bit long, I won't mind if you prefer to keep the current syntax :)

Reply:
Yeah, it's a bit long. I prefer to keep the  `data`.